lib/queue.js

```diff
@@ -61,6 +63,19 @@ var Queue = function Queue(name, redisPort, redisHost, redisOptions){
     redisOptions.db = redisOpts.DB;
   }
 
+  if (_.isString(redisPort)) {
+    try {
+      var redisUrl = url.parse(redisPort);
+      assert(_.isObject(redisHost) || _.isUndefined(redisHost));
```

Comment:
`assert` takes a second argument for the error message so I think we should pass a proper error message instead of assertion failure message.